### DISCUSSION TITLE
Does self-hosted runner clear all previous actions during "Set up?"

### DISCUSSION THREAD

**Discussion by rbellamy** [community/community#26143] [Actions]
<p>I am running the self-hosted runner from <a class="mention" href="/myoung34">@myoung34</a> at <a href="https://github.com/myoung34/docker-github-actions-runner" rel="nofollow noopener">myoung34/docker-github-actions-runner</a>.</p>
<p>I am using a shared storage location at /tmp/github-runnner so all containers running the self-hosted actions runner use that directory as their RUNNER_WORKDIR.</p>
<p>It seems that during workflow setup, the /tmp/github-runner/_actions directory is cleared of all previous actions and set with the new, so if two self-hosted runners execute at the same time, the “last one wins.”</p>

**Comment by rbellamy** [comment]
<p>Each container is launched like so:</p>
<pre><code>docker run -d --restart always --name github-runner \
  -e REPO_URL="https://github.com/myoung34/repo" \
  -e RUNNER_NAME="foo-runner" \
  -e RUNNER_TOKEN="footoken" \
  -e RUNNER_WORKDIR="/tmp/github-runner" \
  -v /var/run/docker.sock:/var/run/docker.sock \
  -v /tmp/github-runner:/tmp/github-runner \
  myoung34/github-runner:latest
</code></pre>

**Comment by rbellamy** [comment, marked as answer]
<p>The answer is “Yes.”</p>
<p>From <a href="https://github.com/actions/runner/blob/master/src/Runner.Worker/ActionManager.cs#L62" rel="nofollow noopener">ActionManager.cs#L62</a>:</p>
<pre><code>// Clear the cache (local runner)
IOUtil.DeleteDirectory(HostContext.GetDirectory(WellKnownDirectory.Actions), executionContext.CancellationToken);
</code></pre>
<p>This means that only one runner can use the RUNNER_WORKDIR at a time.</p>

**Reply by ErcinDedeoglu** [reply]
Is this mean a self-hosted runner cannot execute parallel jobs?
